src/imageResource.js

```diff
@@ -3,46 +3,46 @@
 require('openseadragon');
 
 var ImageResource = function(config) {
+  if(!config) {
+    return;
+  }
   this.needed = config.needed || false;
   this.visible = config.visible || false;
   this.clipRegion = config.clipRegion;
   this.opacity = config.opacity || 1;
-  this.x = config.x || 0;
-  this.y = config.y || 0;
-  this.height = config.height || 1;
-  this.width = config.width || 1;
+  this.bounds = config.bounds || new OpenSeadragon.Rect(0, 0, 1, 1);
   this.zIndex = config.zIndex || 0;
   this.tileSource = config.tileSource;
   this.dynamic = config.dynamic || false;
-  this.imageType = config.imageType || "main"; // can be 'main', 'alternate', or 'detail'
+  this.imageType = config.imageType || "main"; // can be 'main', 'alternate', 'detail' or 'thumbnail'
   this.status = 'initialized'; // can be 'requested', 'received', 'pending','shown', or 'failed'
   this.parent = config.parent;
-  this.dispatcher = config.dispatcher;
+  this.dispatcher = config.parent.dispatcher;
+  this.viewer = config.parent.viewer;
 };
 
 ImageResource.prototype = {
   hide: function() {
-    this.previousOpacity = this.opacity;
     this.visible = false;
-    this.updateOpacity(0);
+    if(this.tiledImage) {
+      this.tiledImage.setOpacity(0);
+    }
   },
 
   show: function() {
     this.visible = true;
-    this.updateOpacity(this.opacity);
+    this.updateOpacity();
   },
 
-  updateOpacity: function(opacity) {
-    if(this.tiledImage) {
-      this.tiledImage.setOpacity(opacity * this.parent.getOpacity());
+  updateOpacity: function() {
+    if(this.visible && this.tiledImage) {
+      this.tiledImage.setOpacity(this.opacity * this.parent.getOpacity());
```

Comment:
You should always setOpacity here (as long as there's a tiledImage). If `!visible` set it to 0. 

Otherwise things will suddenly become visible again when the CanvasObject calls updateOpacity when you're hidden.